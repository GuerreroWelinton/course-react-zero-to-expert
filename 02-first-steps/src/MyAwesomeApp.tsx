// export function MyAwesomeApp() {
//   return (
//     <>
//       <h1>Welinton</h1>
//       <h3>Guerrero</h3>
//     </>
//   );
// }

import type { CSSProperties } from 'react';

const firstName = 'Welinton';
const lastName = 'Guerrero';

const favoriteGames = ['Fornite', 'Smite', 'Minecraft', 'Halo'];
const isActive = false;

const address = {
  zipCode: 'ABC',
  country: 'Ecuador',
};

const myStyles: CSSProperties = { backgroundColor: '#0c5a02ff', borderRadius: '8px', padding: '10px' };

export const MyAwesomeApp = () => {
  return (
    <div data-testid="div-app">
      <h1 data-testid="first-name-title"> {firstName} </h1>
      <h3> {lastName} </h3>

      <p>{2 + 2}</p>
      <p className="mi-clase-favorita"> {favoriteGames.join(', ')} </p>

      <h1> {isActive ? 'Active' : 'Inactive'} </h1>

      <p style={myStyles}> {JSON.stringify(address)} </p>
    </div>
  );
};

// Los valores booleanos (true, false), null y undefined no producen ninguna salida visual en React. Esto es Ãºtil para el renderizado condicional.
// No se pueden imprimir directamente los objetos, pero se puede hacer con un JSON.stringify(object)
