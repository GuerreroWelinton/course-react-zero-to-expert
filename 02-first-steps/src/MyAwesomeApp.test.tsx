import { render, screen } from '@testing-library/react';
import { describe, expect, test } from 'vitest';
import { MyAwesomeApp } from './MyAwesomeApp';

describe('MyAwesomeApp', () => {
  test('should render firstName and lastName', () => {
    const { container } = render(<MyAwesomeApp />);
    // screen.debug();

    const h1 = container.querySelector('h1');
    const h3 = container.querySelector('h3');

    // expect(h1?.innerHTML).toBe('Welinton');
    // expect(h1?.innerHTML).toBe(' Welinton ');
    // expect(h3?.innerHTML).toContain('gue');

    expect(h1?.innerHTML).toContain('Welinton');
    expect(h3?.innerHTML).toContain('rrero');
  });

  test('should render firstName and lastName - screen ', () => {
    render(<MyAwesomeApp />);
    // screen.debug();

    // const h1 = screen.getByRole('heading', {
    //   level: 1,
    // });

    const h1 = screen.getByTestId('first-name-title');
    expect(h1.innerHTML).toContain('Welinton');
  });

  test('should match snapshot', () => {
    const { container } = render(<MyAwesomeApp />);
    expect(container).toMatchSnapshot();
  });

  test('should match snapshot', () => {
    render(<MyAwesomeApp />);
    expect(screen.getByTestId('div-app')).toMatchSnapshot();
  });
});

// Se recomienda solo usar { container } con un render inicial y para estados sin manipulaciones sin eventos
// Se debe actualizar la configuración de vite (vite.config.ts) para poder usar el render
// El screen.debug() solo funciona si hago uso del render(<MyComponent/>)
// Se pueden poner tantos expect dentro del test como se necesiten
// El screen.debug() me permite ver los datos actualizados si surgió un cambio en el componente
// No se recomienda hacerlo con screen.getByTestId('id')
// Los snapshots son fotografías de un componente renderizado en cierto momento, se utliza cuando se quiere verificar
// que la estructura física del componente creado es exactamente igual a la usada previamente
// Si un componente recibe muchas modificaciones es mejor no usar snapshot
