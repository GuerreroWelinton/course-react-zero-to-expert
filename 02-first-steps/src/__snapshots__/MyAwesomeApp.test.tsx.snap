// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshoot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
       
      Welinton
       
    </h1>
    <h3>
       
      Guerrero
       
    </h3>
    <p>
      4
    </p>
    <p
      class="mi-clase-favorita"
    >
       
      Fornite, Smite, Minecraft, Halo
       
    </p>
    <h1>
       
      Inactive
       
    </h1>
    <p
      style="background-color: rgb(12, 90, 2); border-radius: 8px; padding: 10px;"
    >
       
      {"zipCode":"ABC","country":"Ecuador"}
       
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > should match snapshoot 2`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
     
    Welinton
     
  </h1>
  <h3>
     
    Guerrero
     
  </h3>
  <p>
    4
  </p>
  <p
    class="mi-clase-favorita"
  >
     
    Fornite, Smite, Minecraft, Halo
     
  </p>
  <h1>
     
    Inactive
     
  </h1>
  <p
    style="background-color: rgb(12, 90, 2); border-radius: 8px; padding: 10px;"
  >
     
    {"zipCode":"ABC","country":"Ecuador"}
     
  </p>
</div>
`;

exports[`MyAwesomeApp > should match support 1`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
     
    Welinton
     
  </h1>
  <h3>
     
    Guerrero
     
  </h3>
  <p>
    4
  </p>
  <p
    class="mi-clase-favorita"
  >
     
    Fornite, Smite, Minecraft, Halo
     
  </p>
  <h1>
     
    Inactive
     
  </h1>
  <p
    style="background-color: rgb(12, 90, 2); border-radius: 8px; padding: 10px;"
  >
     
    {"zipCode":"ABC","country":"Ecuador"}
     
  </p>
</div>
`;
